Code/GraphMol/SmilesParse/SmilesWrite.cpp

```diff
@@ -595,10 +608,15 @@ std::string MolToSmiles(const ROMol &mol, const SmilesWriteParams &params) {
       }
     }
   }
-  mol.setProp(common_properties::_smilesAtomOutputOrder, flattenedAtomOrdering,
-              true);
-  mol.setProp(common_properties::_smilesBondOutputOrder, flattenedBondOrdering,
-              true);
+  {
+#ifdef RDK_THREADSAFE_SSS
+      std::scoped_lock locker(smiles_setprop_locker());
```

Comment:
Curiosity question. Couldn't this just be:
```suggestion
      static std::mutex proplock;
      std::scoped_lock locker(proplock);
```

Reply:
I’m sharing the lock between moltosmiles and MolFragmentToSmiles in case the same mol goes to both.

Reply:
This should be tested as well for safety.

Reply:
> I’m sharing the lock between moltosmiles and MolFragmentToSmiles in case the same mol goes to both.

Ah. Right. Good point. 
We should at some point do some refactoring so that there's less duplicated code in those functions, but that's clearly not for this PR.

Reply:
@greglandrum I think it's pretty safe but I can add the test this weekend if you like.

Reply:
@bp-kelley, nope, I don't think it's necessary.